### ISSUE TITLE
Write track randomization data to XML instead of ID3 tags

### ISSUE THREAD

**Issue by a-rich** [a-rich/DJ-Tools#4]
**Update:**
1. Feature branch [xml-track-randomization](https://github.com/a-rich/DJ-Tools/tree/xml-track-randomization) created.
2. [Feature merged into 2.1.0 dev branch](https://github.com/a-rich/DJ-Tools/commit/7230c9b377edc4855aa778b96ee290bad27f9f70).
3. #12 merged feature into the 2.1.0 beta release `pip install "dj-beatcloud[levenshtein]" --pre`

**Is your feature request related to a problem? Please describe.**
When using the [randomize_tracks](https://github.com/a-rich/DJ-Tools/blob/main/src/djtools/utils/randomize_tracks.py) module, the mp3 files themselves have their `track_num` ID3 tag modified and the user is expected to `Reload Tags` in Rekordbox. The problem with this is that reloading tags also restores potentially unwanted data that you've modified in Rekordbox but doesn't get saved to the ID3 field; for example, comments often have urls and whatever else in the comments which I delete and replace with more useful info.

[Relevant note in README](https://github.com/a-rich/DJ-Tools/blob/2.1.0/README.md#reloading-tags).

**Describe the solution you'd like**
The `randomize_tracks` module should instead generate a `randomize_rekordbox.xml`, based on `XML_PATH`, which the user then imports tracks from.

**Comment by a-rich** [comment]
Track randomization via Rekordbox tags, instead of MP3 ID3 tags, is sort of kind of working.

Known issues:
1. I saw duplicate numbers in the one playlist I tested with...need to debug to find out why that might be happening.
2. `TrackNumber` assignment is done on a playlist-level rather than globally across all playlists being shuffled...this means that tracks that may be mutually exclusive across the playlists being randomized could end up having the same `TrackNumber` assignment...the [list(range(len(tracks)))](https://github.com/a-rich/DJ-Tools/blob/7230c9b377edc4855aa778b96ee290bad27f9f70/src/djtools/utils/randomize_tracks.py#L57) call should be replaced with list of track numbers looked up in a track -> number dict generated prior to [this for-loop](https://github.com/a-rich/DJ-Tools/blob/7230c9b377edc4855aa778b96ee290bad27f9f70/src/djtools/utils/randomize_tracks.py#L47-L61).
